python_modules/dagster-graphql/dagster_graphql/implementation/loader.py

```diff
@@ -0,0 +1,130 @@
+from collections import defaultdict
+from enum import Enum
+
+from dagster import DagsterInstance, check
+from dagster.core.host_representation import ExternalRepository
+from dagster.core.scheduler.instigation import InstigatorType
+from dagster.core.storage.pipeline_run import JobBucket, TagBucket
+from dagster.core.storage.tags import SCHEDULE_NAME_TAG, SENSOR_NAME_TAG
+
+
+class RepositoryDataType(Enum):
+    JOB_RUNS = "job_runs"
+    SCHEDULE_RUNS = "schedule_runs"
+    SENSOR_RUNS = "sensor_runs"
+    SCHEDULE_STATES = "schedule_states"
+    SENSOR_STATES = "sensor_states"
+
+
+class RepositoryScopedBatchLoader:
+    def __init__(self, instance, external_repository):
+        self._instance = check.inst_param(instance, "instance", DagsterInstance)
+        self._repository = check.inst_param(
+            external_repository, "external_repository", ExternalRepository
+        )
+        self._data = {}
```

Comment:
typing for this for example i think would be really useful for understanding this beast